gnocchi/indexer/sqlalchemy.py

```diff
@@ -664,6 +664,35 @@ def create_metric(self, id, creator, archive_policy_name,
             raise
         return m
 
+    def copy_metric(self, id, creator, archive_policy_name, source_metric):
```

Comment:
This really looks like `create_metric()` actually. I'm not sure you really need this function in the indexer, the job can be done by `gnocchi.rest` directly.

At least if you need it, just use `create_metric()` here.